src/Facets/StargateFacetV2.sol

```diff
@@ -0,0 +1,167 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.17;
+
+import { ILiFi } from "../Interfaces/ILiFi.sol";
+import { IStargate, ITokenMessaging } from "../Interfaces/IStargate.sol";
+import { LibAsset } from "../Libraries/LibAsset.sol";
+import { LibDiamond } from "../Libraries/LibDiamond.sol";
+import { ReentrancyGuard } from "../Helpers/ReentrancyGuard.sol";
+import { InformationMismatch, AlreadyInitialized, NotInitialized, ContractCallNotAllowed, InvalidCallData } from "../Errors/GenericErrors.sol";
+import { SwapperV2, LibSwap } from "../Helpers/SwapperV2.sol";
+import { Validatable } from "../Helpers/Validatable.sol";
+import { ERC20, SafeTransferLib } from "solmate/utils/SafeTransferLib.sol";
+
+/// @title StargateFacetV2
+/// @author Li.Finance (https://li.finance)
+/// @notice Provides functionality for bridging through Stargate (V2)
+/// @custom:version 1.0.0
+contract StargateFacetV2 is ILiFi, ReentrancyGuard, SwapperV2, Validatable {
+    using SafeTransferLib for ERC20;
+
+    /// STORAGE ///
+    ITokenMessaging public immutable tokenMessaging;
+
+    /// @param assetId The Stargate-specific assetId for the token that should be bridged
+    /// @param sendParams Various parameters that describe what needs to be bridged, how to bridge it and what to do with it on dst
+    /// @param fee Information about the (native) LayerZero fee that needs to be sent with the tx
+    /// @param refundAddress the address that is used for potential refunds
+    struct StargateData {
+        uint16 assetId;
+        IStargate.SendParam sendParams;
+        IStargate.MessagingFee fee;
+        address payable refundAddress;
+    }
+
+    /// ERRORS ///
+    error InvalidAssetId(uint16 invalidAssetId);
+
+    /// CONSTRUCTOR ///
+    /// @param _tokenMessaging The address of the tokenMessaging contract (used to obtain pool addresses)
+    constructor(address _tokenMessaging) {
+        tokenMessaging = ITokenMessaging(_tokenMessaging);
+    }
+
+    /// EXTERNAL METHODS ///
+
+    /// @notice Bridges tokens via Stargate Bridge
+    /// @param _bridgeData Data used purely for tracking and analytics
+    /// @param _stargateData Data specific to Stargate Bridge
+    function startBridgeTokensViaStargate(
+        ILiFi.BridgeData calldata _bridgeData,
+        StargateData calldata _stargateData
+    )
+        external
+        payable
+        nonReentrant
+        refundExcessNative(payable(msg.sender))
```

Comment:
Isn't in this case preferrable to use `tx.origin`? Depends on the desired outcome of course, if the `msg.sender` is a contract that's unable to `receive()`, the refund will result in a rejected transaction.

Reply:
Thanks for the comment. We never use tx.origin as this has other undesired implications. 
Most contracts that interact with our contract will have a receive function.
Closing this conversation as it's a known issue